static/app/views/replays/detail/memoryChart.tsx

```diff
@@ -0,0 +1,95 @@
+import styled from '@emotion/styled';
+
+import {AreaChart, AreaChartProps} from 'sentry/components/charts/areaChart';
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {formatBytesBase2} from 'sentry/utils';
+import theme from 'sentry/utils/theme';
+
+function MemoryChart({memorySpans}) {
+  const chartOptions: Omit<AreaChartProps, 'series'> = {
+    grid: {
+      left: '10px',
+      right: '10px',
+      top: '40px',
+      bottom: '10px',
+    },
+    tooltip: {
+      trigger: 'axis',
+      valueFormatter: (value: number | null) => formatBytesBase2(value || 0),
+    },
+    xAxis: {
+      type: 'time',
+      min: memorySpans[0]?.data?.timestamp,
+      max: memorySpans.slice(-1)?.data?.timestamp,
+    },
+    yAxis: {
+      type: 'value',
+      name: t('Heap Size'),
+      nameTextStyle: {
+        padding: 8,
+        fontSize: theme.fontSizeLarge,
+        fontWeight: 600,
+        lineHeight: 1.2,
+        color: theme.gray100,
+      },
+      min: 0,
+      max: memorySpans.slice(-1)?.data?.memory.totalJSHeapSize,
```

Comment:
What happens if the `max` isn't set? 

Rounding up to the nearest 10's should make the scale on the screen nicer, better to see `60MiB` instead of `57.2MiB` in my opinion. 
Maybe this is better as a follow up PR, where we set the scale based on the magnitude of the max value, so sometimes round to the nearest 10's, but maybe 50's or 100 if more memory is used

Reply:
I'm actually not setting it, it seems like echarts interpolates based on the data input, so the grid looks the same.

Reply:
I'll add it to this ticket. https://github.com/getsentry/sentry/issues/34110